scipy/stats/_stats_py.py

```diff
@@ -999,51 +1003,68 @@ def moment(a, order=1, axis=0, nan_policy='propagate', *, center=None):
     2.0
 
     """
-    moment = order  # parameter was renamed
-    a, axis = _chk_asarray(a, axis)
+    xp = array_namespace(a)
+    a, axis = _chk_asarray(a, axis, xp=xp)
 
-    # for array_like moment input, return a value for each.
-    if not np.isscalar(moment):
+    if xp.isdtype(a.dtype, 'integral'):
+        a = xp.asarray(a, dtype=xp.float64)
+    else:
+        a = xp.asarray(a)
+
+    order = xp.asarray(order, dtype=a.dtype)
+    if xp_size(order) == 0:
+        # This is tested by `_moment_outputs`, which is run by the `_axis_nan_policy`
+        # decorator. Currently, the `_axis_nan_policy` decorator is skipped when `a`
+        # is a non-NumPy array, so we need to check again. When the decoratpor is
```

Comment:
```suggestion
        # is a non-NumPy array, so we need to check again. When the decorator is
```

Reply:
Lumped this in with other changes locally rather than committing here.